### DISCUSSION TITLE
Best approach for passing dynamic data to navbar and footer using getStaticProps

### DISCUSSION THREAD

**Discussion by stijnwesterhof** [vercel/next.js#14008] [Help]
I used the getStaticProps for pre-rendering my product pages and it works really well, I like it! But I'm currently thinking of a way how I can use it (or if I should use it) for my navbar and footer.

The links that are going to be placed in the navbar and footer are dynamic and manageable via a Headless CMS. I know I can only use getStaticProps on pages/index files. So should I fetch the data for a navbar on every page's getStaticProps or is there a better way? Like a wrapper for example?

I hope someone can help me with this,

**Comment by lfades** [comment]
> So should I fetch the data for a navbar on every page's getStaticProps or is there a better way?

TL;DR: In the page.

You could create a helper function to make it easier, and even add React Context if the components are very nested. We don't support `getStaticPaths` / `getStaticProps` in nested components unfortunately, only in pages.
